platform_data/defs.bzl

```diff
@@ -0,0 +1,86 @@
+"""
+The platform_data rule can be used to change the target platform of a target,
+and then depend on that elsewhere in the build tree. For example:
+
+cc_binary(name = "foo")
+
+platform_data(
+    name = "foo_embedded",
+    target = ":foo",
+    platform = "//my/new:platform",
+)
+
+py_binary(
+    name = "flasher",
+    srcs = ...,
+    data = [
+        ":foo_embedded",
+    ],
+)
+
+Regardless of what platform the top-level :flasher binary is built for,
+the :foo_embedded target will be built for //my/new:platform. 
+
+Note that if you depend on :foo_embedded it's not exactly the same as depending on :foo, since it won't forward all the same providers. In the future, we can extend this to add some common providers as needed"""
+
+def _target_platform_transition_impl(attr):
+    return {
+        "//command_line_option:platforms": str(attr.platform),
```

Comment:
isn't this supposed to be `str` and the option has the label converter? https://github.com/bazelbuild/bazel/blob/ea76dd25bd843ce8d475c06be2307341e407b1fe/src/main/java/com/google/devtools/build/lib/analysis/PlatformOptions.java#L85

Reply:
I'm not sure what actually works: it'd be nice if the attribute was an `attr.label` and stayed an actual `Label` all the way, instead of being round-tripped through a String.

Reply:
the rule's attribute is actually a label and this is the transition setting the flag so i think having it as string is okay considering the `Label` converter the option already has. Also looking at a similar transition that set platform, `str` is being used [here](https://github.com/aspect-build/bazel-lib/blob/f2bb1d1519390f04b7974b44c1c6879b1ffd33bd/lib/transitions.bzl#L6) as well